src/api/options.c

```diff
@@ -157,10 +163,14 @@ nlopt_opt NLOPT_STDCALL nlopt_copy(const nlopt_opt opt)
             nopt->xtol_abs = (double *) malloc(sizeof(double) * (opt->n));
             if (!opt->xtol_abs)
                 goto oom;
+            nopt->x_weights = (double *) malloc(sizeof(double) * (opt->n));
```

Comment:
I'd rather have these initialized to NULL unless they are actually used — i.e. in the common case where the weights are equal they are not allocated.  But I guess we should do the same thing for `xtol_abs` as well, so that could be done in a separate PR.